server/controllers/items/recent_public.coffee

```diff
@@ -3,29 +3,42 @@ _ = __.require 'builders', 'utils'
 error_ = __.require 'lib', 'error/error'
 items_ = __.require 'controllers', 'items/lib/items'
 bundleOwnersToItems = require './lib/bundle_owners_to_items'
+itemsQueryLimit = 100
+maxItemsPerOwner = 3
+offset = 0
 
 module.exports = (req, res)->
   { query } = req
-  assertImage = query['assert-image'] is 'true'
-  reqUserId = req.user?._id
+  { limit, lang } = query
+
+  if limit?
+    unless _.isPositiveIntegerString limit
+      return error_.bundleInvalid req, res, 'limit', limit
+
+    limit = parseInt limit
+
+  limit or= 15
 
-  itemsQueryLimit = 100
-  offset = 0
-  itemsReturnedLimit = 15
-  maxItemsPerOwner = 3
+  if lang? and not _.isLang(lang)
+    return error_.bundleInvalid req, res, 'lang', lang
+
+  lang or= 'en'
+
+  assertImage = _.parseBooleanString query['assert-image']
+  reqUserId = req.user?._id
 
-  items_.publicByDate itemsQueryLimit, offset, assertImage, reqUserId
-  .then selectRecentItems.bind(null, itemsReturnedLimit, maxItemsPerOwner)
+  items_.publicByLangAndDate itemsQueryLimit, offset, lang, assertImage, reqUserId
+  .then selectRecentItems(limit, maxItemsPerOwner)
```

Comment:
why changing the limit name ? `itemsReturnedLimit` is supposed to make a clear distinction with the other limit of this function `itemsQueryLimit`

Reply:
I'm re-using the parameter's name as it is now passed in the request query. Given the limit is defined by the request, it's purpose to be used as the items returned limit seems quite clear. We could have a short line of comment on the line 30 to explicit why we need to query 100 before selecting, to make the distinction between those 2 limits even more clear